README.md

```diff
@@ -8,4 +8,249 @@ are in the process of being changed and will certainly break your code.
+Create a new context keeping the old tag set and adding new tags to it, removing specific tags from it, or modifying the values fo some tags.
+    
+    oldTagSet := tags.FromContext(ctx)
+    newTagSet := &tags.TagSetBuilder{}.StartFromTagSet(oldTagSet)
+                                      .InsertString( key1, "foo value")
+                                      .UpdateString( key1, "foo value2")
+                                      .UpsertString( key2, "bar value")
+                                      .Build()
+    ctx2 := tags.ContextWithChanges(ctx, newTagSet)
+
+## Registering views and retrieving their collected data.
+
+### To create/register a measure a.k.a resource
+Create/load measures units:
+
+    // returns a *MeasureFloat64
+    mf := stats.NewMeasureFloat64("/my/float64/measureName", "some measure")
+    mi := stats.NewMeasureInt64("/my/otherName", "some other measure")
+    ...
+
+Register measures:
+
+	if err := stats.RegisterMeasure(mf); err != nil {
+        // handle error
+    }
+    if err := stats.RegisterMeasure(mi); err != nil {
+        // handle error
+    }
+    ...
+
+Retrieve measure by name:
+    mf, err := stats.GetMeasureByName("my/float64/measureName")
+	if err != nil {
+        // handle error
+    }
+    mi, err := stats.GetMeasureByName("/my/otherName")
+    if err != nil {
+        // handle error
+    }
+    ...
+
+Unregister measures:
+	if err := stats.UnregisterMeasure(mf); err != nil {
+        // handle error
+    }
+    if err := stats.UnregisterMeasure(mi); err != nil {
+        // handle error
+    }
+    ...
+
+### To create an aggregation type
+Currently only 2 types of aggregations are supported. The AggregationCount is used to count the number of times a sample was recorded. The AggregationDistribution is used to provide a histogram of the values of the samples.
+
+    histogramBounds := []float64 { -10, 0, 10, 20}
+    agg1 := stats.NewAggregationDistribution(histogramBounds)
+    agg2 := stats.NewAggregationCount()
+
+
+### To create an aggregation window
+Currently only 3 types of aggregation windows are supported. The WindowCumulative is used to continuously aggregate the data received. The WindowSlidingTime to aggregate the data received over the last specified time interval. The NewWindowSlidingCount to aggregate the data received over the last specified sample count.
+Currently all aggregation types are compatible with all aggregation windows. Later we might provide aggregation types that are incompatible with some windows.
+
+    duration := 10 * time.Second
+    precisionIntervals := 5
+    wnd1 := stats.NewWindowSlidingTime(duration, precisionIntervals)
+
+    lastNSamples := 100
+    precisionSubsets := 10
+    wnd2 := stats.NewWindowSlidingCount(lastNSamples, precisionSubsets)
+    
+    wn3 := stats.NewWindowCumulative()
+
+
+### To creater/register a view
+Create a view:
+
+    vw1 = stats.NewViewInt64("my/int64/viewName", "some description", []Keys{key1, key2}, mf, agg1, wnd1)
```

Comment:
I would strongly prefer some kind of builder pattern here. We have too many parameters for this call.

Reply:
I don't see the benefits. 
The construction logic of a view is straightforward. Why try to encapsulate it?

The types of the arguments are all different (besides the name/description) and they are all required (besides the description). It seems we'll catch more errors this way at compile time vs using a builder pattern where users can forget to set arguments and failures won't occur until runtime.

Reply:
I just don't like constructors with >3 arguments but I see your point.

Reply:
Understood. But for now I prefer the safety of compile time error.